src/hotspot/share/gc/serial/cardTableRS.cpp

```diff
@@ -601,20 +345,188 @@ void CardTableRS::verify() {
 CardTableRS::CardTableRS(MemRegion whole_heap) :
   CardTable(whole_heap) { }
 
+// Implemented word-iteration to skip long consecutive clean cards.
+CardTable::CardValue* CardTableRS::find_first_dirty_card(CardValue* const start_card,
+                                                         CardValue* const end_card) {
+  using Word = uintptr_t;
+
+  static_assert(clean_card_val() == (CardValue)-1, "inv");
+  constexpr Word clean_word = (Word)CardTable::clean_card_row_val();
+
+  CardValue* i_card = start_card;
+
+  while (!is_aligned(i_card, sizeof(Word))) {
+    if (i_card >= end_card) {
+      return end_card;
+    }
+    if (is_dirty(i_card)) {
+      return i_card;
+    }
+    ++i_card;
+  }
+
+  // Word comparison
+  while (i_card + sizeof(Word) <= end_card) {
+    Word* i_word = reinterpret_cast<Word*>(i_card);
+    if (*i_word != clean_word) {
```

Comment:
Why not just use clean_card_row_val() here directly? Rather than introducing the clean_word variable and
only using it once, here.